app/controllers/courses_controller.rb

```diff
@@ -312,6 +341,14 @@ def verify_edit_credentials
     yield
   end
 
+  # Makes sure that the home wiki
+  # is always a part of courses wikis.
+  def ensure_home_wiki_in_courses_wikis
```

Comment:
I was misunderstanding the place you had implemented this validation; I thought it was done on the model, with a `before_save` callback on the model. Doing it here doesn't give us much extra assurance beyond the frontend validation; it won't save us from mistakes elsewhere in the code that create or update a course but fail to include the home_wiki.

Reply:
Hm that's a valid point. I'll patch it up.